### PR TITLE
Ztx enable disable shutdown

### LINKED ISSUE #110: issuing ziti_shutdown on windows produces Assertion failure
Steps to reproduce:
* connect an identity 
* verify things work properly
* try to reclaim memory/remove identity using `ziti_shutdown`
* observe: 
```
Assertion failed!

Program: c:\git\github\openziti\ziti-tunnel-win\service\ziti-tunnel.exe
File: ../deps/ziti-sdk-c/deps/uv-mbed/deps/libuv/src/win/async.c, Line 76

Expression: !(handle->flags & UV_HANDLE_CLOSING)
```

### LINKED ISSUE #301: Implement disabling/enabling of Ziti context
Here is the things that happen on disable:

- active `ziti_connection`s are disconnected, appropriate callback receives `ZITI_DISABLED` error code
- `ZitiServiceEvent` is sent with all current services removed
- connections to Edge Routers are closed, with corresponding `ZitiEdgeRouteEvent`s
- ziti_context terminates API session (controller /logout), with corresponding `ZitiContextEvent`
- all model data (services, sessions, api session) is freed

if context is disabled it can be re-enabled/restarted by calling `ziti_set_enabled(ztx, true)`. Re-enabling puts it back into 
running state

### PR DESCRIPTION
**Pull request by ekoby**
[fixes #301]
[fixes #110]

### CODE REVIEW

File: library/ziti.c [hunk 1]
```diff
@@ -206,14 +212,72 @@ int ziti_init_opts(ziti_options *options, uv_loop_t *loop) {
     return ZITI_OK;
 }
 
-static void ziti_init_async(uv_async_t *ar) {
-    struct ziti_init_req *init_req = ar->data;
+extern bool ziti_is_enabled(ziti_context ztx) {
+    return ztx->enabled;
```
Comment (dovholuknf) [inline library/ziti.c:216]: should this guard against a uninitialized/null'ed context? it'd be friendlier to not crash imo... but maybe that's just me? :)

Comment (ekoby) [reply library/ziti.c:216]: just you

otherwise we need to do it everywhere

File: includes/ziti/error_defs.h [hunk 2]
```diff
@@ -87,7 +87,9 @@ limitations under the License.
     /** attempted to verify or retrieve details of an MFA enrollment that has not been completed */ \
     XX(MFA_NOT_ENROLLED, "the current identity has not completed MFA enrollment") \
     /** not found, usually indicates stale reference or permission */    \
-    XX(NOT_FOUND, "entity no longer exists or is no longer accessible") \
+    XX(NOT_FOUND, "entity no longer exists or is no longer accessible")  \
+    /** operation attempted while ziti_context is not enabled */                    \
+    XX(DISABLED, "ziti context is disabled") \
```
Comment (dovholuknf) [inline includes/ziti/error_defs.h:92]: i assume `ZITI_OK` is the antithesis of `DISABLED`?

Comment (ekoby) [reply includes/ziti/error_defs.h:92]: ZITI_OK is general _no error_ result

File: library/connect.c [hunk 3]
```diff
@@ -488,6 +488,8 @@ static void ziti_connect_async(uv_async_t *ar) {
 }
 
 static int do_ziti_dial(ziti_connection conn, const char *service, ziti_dial_opts *dial_opts, ziti_conn_cb conn_cb, ziti_data_cb data_cb) {
+    if (!conn->ziti_ctx->enabled) return ZITI_DISABLED;
```
Comment (dovholuknf) [inline library/connect.c:491]: why not use `ziti_is_enabled` here? same comment below

File: library/ziti.c [hunk 4]
```diff
@@ -206,14 +212,72 @@ int ziti_init_opts(ziti_options *options, uv_loop_t *loop) {
...
+
+        ev.type = ZitiContextEvent;
+        ev.event.ctx.ctrl_status = ZITI_DISABLED;
+
+        ziti_send_event(ztx, &ev);
+    }
+}
+
+static void ziti_start_internal(ziti_context ztx, void *init_req) {
+    if (!ztx->enabled) {
```
Comment (dovholuknf) [inline library/ziti.c:269]: personally, i think this could use some log statements. this might be hard to debug/diagnose without.

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**